### PR TITLE
fix for issue 305

### PR DESCRIPTION
**Pull request by sudiptoguha**
*Issue #, if available:* Issue 305. The most recent changes to main broke the state classes older models (version 2.0.1). 

*Description of changes:* Changes the PointStoreMapper to address the transformation and adds tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreMapper.java [hunk 1]
```diff
@@ -55,7 +55,12 @@ public PointStore toModel(PointStoreState state, long seed) {
         int[] locationList = new int[indexCapacity];
         int[] tempList = ArrayPacking.unpackInts(state.getLocationList(), state.isCompressed());
         if (!state.getVersion().equals(Version.V3_0)) {
-            System.arraycopy(tempList, 0, locationList, 0, tempList.length);
+            int shingleSize = state.getShingleSize();
+            int baseDimension = dimensions / shingleSize;
+            for (int i = 0; i < tempList.length; i++) {
+                checkArgument(tempList[i] % baseDimension == 0, " incorrect state");
```
Comment (jotok) [inline Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreMapper.java:61]: Can we provide a more descriptive error message?

Comment (ylwu-amzn) [reply Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreMapper.java:61]: agree, not so easy to get what's wrong with the general error message

Comment (sudiptoguha) [reply Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreMapper.java]: Done

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks a lot for fixing this quickly!

**Review by jotok — APPROVED** [review]

**Pull request merged**